application-services/custom/camera-management/appcamera/commands.go

```diff
@@ -76,42 +66,65 @@ func (app *CameraManagementApp) doPTZ(deviceName, profileToken string, x, y, zoo
 	return app.sendPutCommand(deviceName, relativeMoveCommand, cmd)
 }
 
-func (app *CameraManagementApp) getPresets(deviceName string, profileToken string) (GetPresetsResponse, error) {
-	cmd := &ptz.GetPresets{
-		ProfileToken: onvif.ReferenceToken(profileToken),
-	}
-
-	presets, err := app.issueGetCommand(context.Background(), deviceName, getPresetsCommand, cmd)
+func (app *CameraManagementApp) getCameraFeatures(deviceName string) (CameraFeatures, error) {
+	var err error
+	features := CameraFeatures{}
+	dev, err := app.getDeviceByName(deviceName)
 	if err != nil {
-		return GetPresetsResponse{}, errors.Wrapf(err, "failed to issue get presets command")
+		return CameraFeatures{}, err
 	}
 
-	val := presets.Event.Readings[0].ObjectValue
-	js, err := json.Marshal(val)
-	if err != nil {
-		return GetPresetsResponse{}, errors.Wrapf(err, "failed to marshal presets json object")
+	switch dev.ServiceName {
+	case app.config.AppCustom.OnvifDeviceServiceName:
+		features.CameraType = Onvif
+		caps, err := app.getCapabilities(deviceName)
+		if err != nil {
+			return CameraFeatures{}, errors.Wrapf(err, "unable to get device capabilities")
+		}
+		if caps.Capabilities.PTZ.XAddr != "" {
+			features.PTZ = true
+
+			if ptzConfigs, err := app.getPTZConfiguration(deviceName); err != nil {
+				app.lc.Errorf("Error calling Get PTZ Configuration for device %s: %s", deviceName, err.Error())
+			} else {
+				features.Zoom = ptzConfigs.PTZConfiguration[0].ZoomLimits != nil
+			}
+		}
+	case app.config.AppCustom.USBDeviceServiceName:
+		features.CameraType = USB
+	default:
+		features.CameraType = Unknown
 	}
-	pr := GetPresetsResponse{}
-	err = json.Unmarshal(js, &pr)
-	return pr, err
-}
 
-func (app *CameraManagementApp) getPTZConfiguration(deviceName string) (GetPTZConfigurationsResponse, error) {
-	cmd := &ptz.GetConfigurations{}
+	return features, nil
+}
 
-	config, err := app.issueGetCommand(context.Background(), deviceName, getConfigurationsCommand, cmd)
-	if err != nil {
-		return GetPTZConfigurationsResponse{}, errors.Wrapf(err, "failed to issue get configurations command")
+func (app *CameraManagementApp) getCapabilities(deviceName string) (device.GetCapabilitiesResponse, error) {
+	cmd := &device.GetCapabilities{
+		Category: onvif.CapabilityCategory("All"),
```

Comment:
When I try to build this, I get an error here: `appcamera/commands.go:105:13: cannot use onvif.CapabilityCategory("All") (constant "All" of type onvif.CapabilityCategory) as type []onvif.CapabilityCategory in struct literal`